source/elastic_law.py

```diff
@@ -0,0 +1,260 @@
+
+        self._dV = dlfn.Measure("dx", domain=self._mesh)
+
+        self._I = dlfn.Identity(self._space_dim)
+        
+    def _strain(self):
+        raise NotImplementedError("You are calling a purely virtual method.")
+    
+    def _dstrain(self):
+        raise NotImplementedError("You are calling a purely virtual method.")
+
+    def volume_scaling(self):
+        raise NotImplementedError("You are calling a purely virtual method.")
+
+    def traction_scaling(self):
+        raise NotImplementedError("You are calling a purely virtual method.")
+
+    def dw_int(self):
+        raise NotImplementedError("You are calling a purely virtual method.")
+    
+    def postprocess_cauchy_stress(self, displacement):
+        raise NotImplementedError("You are calling a purely virtual method.")
+
+
+class Hooke(ElasticLaw):
+    def __init__(self):
+        super().__init__()
+        self.nonlinear = False
+        self.linearity_type = "linear"
+        self.name = "Hooke"
+
+    def set_parameters(self, mesh, C, u, v, solution):
+        super().set_parameters(mesh, C, u, v, solution)
+        
+    def _cauchy_stress(self):
+
+        assert hasattr(self, "_C")
+        assert hasattr(self, "_I")
+
+        return self._C * dlfn.tr(self._strain()) * self._I \
+            + dlfn.Constant(2.0) * self._strain()
+
+    def _strain(self):
+
+        assert hasattr(self, "_u")
+
+        return dlfn.Constant(0.5) * (grad(self._u).T + grad(self._u))
+
+    def _dstrain(self):
+
+        assert hasattr(self, "_v")
+
+        return dlfn.Constant(0.5) * (grad(self._v).T + grad(self._v))
+
+    def volume_scaling(self):
+        return dlfn.Constant(1.0)
+
+    def traction_scaling(self):
+        return dlfn.Constant(1.0)
+
+    def dw_int(self):
+        
+        assert hasattr(self, "_dV")
+
+        return inner(self._cauchy_stress(),self._dstrain()) * self._dV
+
+    def postprocess_cauchy_stress(self, displacement):
+
+        assert hasattr(self, "_C")
+        assert hasattr(self, "_I")
+
+        displacement_gradient = dlfn.grad(displacement)
+        # strain tensor (symbolic)
+        strain = dlfn.Constant(0.5) * (displacement_gradient + \
+            displacement_gradient.T)
+        
+        # dimensionless stress tensor (symbolic)
+        stress = dlfn.Constant(self._C) * \
+            dlfn.inner(self._I, strain) * self._I
+        stress += dlfn.Constant(2.0) * strain
+
+        return stress
+
+
+class StVenantKirchhoff(ElasticLaw):
+    def __init__(self):
+        super().__init__()
+        self.nonlinear = True
+        self.linearity_type = "nonlinear"
+        self.name = "StVenantKirchhoff"
+        
+    def set_parameters(self, mesh, C, u, v, solution):
+        super().set_parameters(mesh, C, u, v, solution)
+        
+        # kinematic variables
+        self._F = self._I + grad(self._solution)
+        self._CG = self._F.T * self._F
+        self._J = dlfn.det(self._F)
```

Comment:
Here you begin to use the variable name which correspond to commonly used symbol. Above you are using `displacement_gradient` instead of the symbol `H`. I propose to use either use one of the schemes thoroughly. You could re-name  `self._C` to `self._elasticity_ratio` and use `self._C` for the right Cauchy-Green tensor.